src/upload/index.ts

```diff
@@ -19,24 +19,30 @@ export const upload = (
     return;
   }
 
-  if (!options.authenticationEndpoint) {
-    respond(true, errorMessages.MISSING_AUTHENTICATION_ENDPOINT, callback);
-    return;
-  }
-
   if (!options.publicKey) {
     respond(true, errorMessages.MISSING_PUBLIC_KEY, callback);
     return;
   }
 
+  if(!uploadOptions.token || !uploadOptions.signature || !uploadOptions.expire) {
```

Comment:
Add individual checks and give specific error message so that user knows what exactly is missing.